delete_me_proxies_example/HumanReadableCleaningSteps.Rmd

```diff
@@ -0,0 +1,118 @@
+4. **Harmonizing to generic fractions.** Using the `ResultSampleFractionText` field, group fractions into 4 types: "filtered", "unfiltered", "particulate", and "unknown". Replace the values in `ResultSampleFractionText` with these new generic fractions. Use "filtered" when `ResultSampleFractionText` was "Dissolved", "Filtered, lab", or "Filtered, field". Use "unfiltered" when `ResultSampleFractionText` was "Total", "Total Recoverable", "Total Recovrble", or "Recoverable". Use "particulate" when `ResultSampleFractionText` was "Suspended". Use "unknown" when `ResultSampleFractionText` was missing or said "None", "Pot. Dissolved", "Filterable", "Non-filterable", or "Non-Filterable (Particle)". If `ResultSampleFractionText` matched none of these things, it just kept `ResultSampleFractionText` as-is (we can use this as a way to catch scenarios we haven't seen since there should only be 4 unique fractions returned). Then, add a column called `flag_fracAmbiguous` that says TRUE when `ResultSampleFractionText` is "unknown".
+5. **Removing duplicates.** We defined a unique observation by having distinct combinations of the following columns: `OrganizationIdentifier`, `MonitoringLocationIdentifier`, `ActivityStartDate`, `ActivityStartTime.Time`, `CharacteristicName`, `ResultSampleFractionText`. If any observation had an exact match of the values in all 6 of these columns, then it was treated as a duplicate and removed.
+
+<br>
+
+There were additional criteria applied specifically to individual contaminant groups. For both HABS and METALS, we kept only surface water sites (removed an observation if the value in `MonitoringLocationTypeName` was "Well"). While the data was only queried for `sampleMedia` of either "water" or "Water", some of the METALS data had activity media that were still sediment. So, for METALS, we did one additional filter to remove any observations where the value in `ActivityMediaName` or the value in `ResultSampleFractionText` was "Bed Sediment".
+
+# Below is the code that does the numbered cleaning steps above
+
+\footnotesize
+
+```{r eval=FALSE}
+# Some cleaning functions
+
+checkBelowDetect <- function(df, true_output, false_output) {  
+  # From J. Murphy
+  ifelse(is.na(df$ResultMeasureValue) & 
+           !is.na(df$DetectionQuantitationLimitMeasure.MeasureValue) &
+           df$ResultDetectionConditionText %in% 
+           c("*Non-detect", "Not Detected", "Present Below Quantification Limit"),
+         true_output, false_output)
+}
```

Comment:
Also, the functional approach here is great and sets folks up for that style of thinking later down the line.

The only big change I would make here is making sure that our function arguments are clearly defined (a la standard package documentation).